<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NNConfigWindow</class>
 <widget class="QMainWindow" name="NNConfigWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1147</width>
    <height>792</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NNConfigWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="StartNNButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>550</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Test NN</string>
    </property>
   </widget>
   <widget class="QCustomPlot" name="NNPlotWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1121</width>
      <height>401</height>
     </rect>
    </property>
   </widget>
   <widget class="QSlider" name="learningRateSlider">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>640</y>
      <width>181</width>
      <height>16</height>
     </rect>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::PreventContextMenu</enum>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="momentumSlider">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>670</y>
      <width>181</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>50</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="LearningRateLCD">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>640</y>
      <width>64</width>
      <height>23</height>
     </rect>
    </property>
    <property name="smallDecimalPoint">
     <bool>false</bool>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLCDNumber" name="MomentumLCD">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>670</y>
      <width>64</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value" stdset="0">
     <double>1.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>640</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Learning Rate:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>670</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Momentum:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>690</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Passes to Smooth Over:</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="memoryLCD">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>690</y>
      <width>64</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value" stdset="0">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QSlider" name="memorySlider">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>690</y>
      <width>641</width>
      <height>20</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>1200</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::NoTicks</enum>
    </property>
    <property name="tickInterval">
     <number>50</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>580</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Final Average Error:</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="avgErrorLCD">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>580</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="digitCount">
     <number>7</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>610</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Min Average Error:</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="minValLCD">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>610</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>610</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Min Pass:</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="minPassLCD">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>610</y>
      <width>51</width>
      <height>23</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>420</y>
      <width>111</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Select DataFile</string>
    </property>
   </widget>
   <widget class="QComboBox" name="activationFunctinoComboBox">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>600</y>
      <width>131</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="hiddenLayersSelector">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>570</y>
      <width>51</width>
      <height>29</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="HiddenLayerLabel">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>560</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of Hidden Layers</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="customTopologyChkBox">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>550</y>
      <width>171</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Use Custom Topology</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>580</y>
      <width>81</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Select Activation Function</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="HiddenNeuronLabel">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>620</y>
      <width>101</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string>Initial Number of Neurons/Hidden Layer</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QSpinBox" name="initialNeuronsinHiddenLayers">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>640</y>
      <width>51</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
   </widget>
   <widget class="QRadioButton" name="txtTrainingDataType">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>410</y>
      <width>106</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>.txt File</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="mySQLTrainingDataSourceType">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>440</y>
      <width>106</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>mySQLTable</string>
    </property>
   </widget>
   <widget class="QLabel" name="HiddenNeuronLabel_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>101</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Select Training Data Source Type</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="setupPushButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>510</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Setup mySQL</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addInputPushButton">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>510</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Input</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addOutputPushButton">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>510</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Output</string>
    </property>
   </widget>
   <widget class="QLabel" name="addInputLabel">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>510</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of Inputs</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="addOutputLabel">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>510</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of outputs</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLCDNumber" name="addInputLCD">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>512</y>
      <width>64</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLCDNumber" name="addOutputLCD">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>512</y>
      <width>64</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="saveBiasesandWeights">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>590</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Biases and Weights</string>
    </property>
   </widget>
   <widget class="QPushButton" name="SelectTarget">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>420</y>
      <width>111</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Select Target</string>
    </property>
   </widget>
   <widget class="QLabel" name="targetmodeLabel">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>420</y>
      <width>101</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Select AI Target Mode</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="OPCSelect">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>420</y>
      <width>141</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Process Control</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="offlineSelect">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>450</y>
      <width>106</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Offline Data</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="mySQLTargetSelect">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>480</y>
      <width>106</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>All Humans</string>
    </property>
   </widget>
   <widget class="QCommandLinkButton" name="commandLinkButton">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>680</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Back To Main Menue</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1147</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
